src/io/file_descriptor.cr

```diff
@@ -11,7 +11,12 @@ class IO::FileDescriptor < IO
   getter fd
 
   def initialize(@fd, blocking = false)
-    @closed = false
+    @closed =
+      {% unless flag?(:win32) %}
+        !fcntl(LibC::F_GETFD) rescue true
```

Comment:
You could avoid raising and immediately catching an exception by using `LibC.fcntl` directly. Or alternatively, implement an non-raising `fctnl?` method.

Reply:
Hence why @RX14 proposed to have a `system_closed?` method that would call `LibC.fcntl()` directly, I guess.

Reply:
Dang, I was just trying to follow the advice from the first review.

RX14 has a good idea, I'll move it down a level into `system_closed?`. I'm just trying to touch as little as possible though.